tests/end_to_end/CairoPrecompiles/Pragma/test_pragma_precompile.py

```diff
@@ -19,35 +26,43 @@ def serialize_cairo_response(cairo_dict: OrderedDict) -> Tuple:
     return tuple(value if value is not None else 0 for value in cairo_dict.values())
 
 
-def serialize_data_type(data_type: dict) -> Tuple:
+def serialize_cairo_inputs(data_type: dict, aggregation_mode: AggregationMode) -> Tuple:
     """
-    Serialize the data type to a tuple
+    Serialize the data type & aggregation_mode to a tuple
     with the same format as the one expected by the Solidity contract.
 
     In solidity, the serialized data type is a tuple with the following format:
-    (entry_type, pair_id, expiration_timestamp)
+    (entry_type, pair_id, expiration_timestamp, aggregation_mode)
       - SpotEntry and GenericEntry take one argument pair_id
       - FutureEntry takes two arguments pair_id and expiration_timestamp
 
     The `expiration_timestamp` is set to 0 for SpotEntry and GenericEntry.
     """
     entry_type, query_args = next(iter(data_type.items()))
     serialized_entry_type = ENTRY_TYPE_INDEX[entry_type]
+    serialized_aggregation_mode = aggregation_mode.value
 
     if isinstance(query_args, tuple):
         pair_id, expiration_timestamp = query_args
-        return (serialized_entry_type, pair_id, expiration_timestamp)
+        return (
+            serialized_entry_type,
+            pair_id,
+            expiration_timestamp,
+            serialized_aggregation_mode,
+        )
```

Comment:
Shouldn't this be ?
```suggestion
        return (
            serialized_aggregation_mode,
            serialized_entry_type,
            pair_id,
            expiration_timestamp,
        )
```

From my understanding, this is the serialized solidity sent to the function `pragma_caller.getData` with the struct to pass being 
```solidity
struct PragmaPricesRequest {
        AggregationMode aggregationMode;
        DataType dataType;
        uint256 pairId;
        uint256 expirationTimestamp;
    }
```

Reply:
Looks to error too, but thanks, I'll dive more into this soon, kinda loaded with other work for today~tomorrow

Reply:
Indeed seems to be something else. Closing this and will check when time today